### PR TITLE
Start for man pages for all scripts except aursift.

### PR DESCRIPTION
**Pull request by runical**
A start for the documentation to solve #76. I would like to get some feedback on the factual accuracy and what would be the best way to get this into groff format. I did not change the documentation for aursift and did not document aursplit. Aurqueue is also sparsely documented.

### CODE REVIEW

File: doc/aurbuild.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+aurbuild - Build packages to a local repository
+
+## Synopsis
+
+aurbuild [+-a ARG] [+-cd ARG] [+-p ARG] [+-r ARG} [--] ARGS...
+
+## Description
+
+Build the packages from a PKGBUILD and add them to a local repository. The PKGBUILD file must be given, just like the name of the database and the directories for the packages and the database files.
```
Comment (AladW) [inline doc/aurbuild.md]: This gives the impression that an actual PKGBUILD should be specified to
aurbuild -a, while it's a text file with directory names containing a
PKGBUILD.

File: doc/aurbuild.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+## Operations
+
+* __-a__ The PKGBUILD files to build
+* __-c__ Build in a chroot
+* __-d__ The name of the database (required)
+* __-p__ Select the location to put the built packages (required)
```
Comment (AladW) [inline doc/aurbuild.md]: I'd probably mention the term "pool", as this is also mentioned in repose(1).

File: doc/aurchain.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+aurchain - Solve and order dependencies using the AUR RPC
+
+## Synopsis
+
+aurchain package [package, ...]
+
+## Description
+
+aurchain takes a list of package names and then solves and orders the dependencies for those packages.
```
Comment (AladW) [inline doc/aurchain.md]: A list of package names implies it takes a file as an argument, however, it
takes package names as command-line arguments directly.

File: doc/aurchain.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+## Operations
+
+* __-v__ Output more
```
Comment (AladW) [inline doc/aurchain.md]: I've removed this for now, but may readd it in future.

File: doc/aursearch.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+## Description
+
+Search the AUR for packages. The argument is a PCRE pattern to search the AUR with.
+
+## Operations
+
+* __-F__ Search for a fixed string instead of a PCRE pattern
+* __-b__ Match pkgbase instead of pkgname
+* __-q__ Output less
+* __-r__ Output raw
```
Comment (AladW) [inline doc/aursearch.md]: Perhaps mention that "raw" is the JSON output from the RPC.

File: doc/aursearch.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+* __-v__ Output more
+
+## See also
+
+__regex__(7)
```
Comment (AladW) [inline doc/aursearch.md]: regex(7) covers BRE and ERE; I think PCRE is covered by pcrepattern(3).

File: doc/aursync.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+aursync - download and build AUR packages automatically
+
+## Synopsis
+
+aursync [+-cdi ARG] [+-knut] [--] ARGS...
+
+## Description
+
+Downloads and builds AUR packages automatically. It asks which repository to use in case of multiple local repositories.
```
Comment (AladW) [inline doc/aursync.md]: Defining what a "local repository" is would be good, i.e., mention the
file:/// prefix.

File: doc/aursync.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+* __-c__ Build packages in a chroot
+* __-d__ Only download source files, don't build anything
+* __-i__ Ignore package. Currently unused
+* __-k__ Don't view downloaded files before building
+* __-n__ Disable version checking for packages in the queue
+* __-u__ Update packages in the custom repository
+* __-t__ Download the PKGBUILDs using aria instead of git
+
+All arguments after -- get passed to makepkg
```
Comment (AladW) [inline doc/aursync.md]: Note that this isn't yet implemented.

File: doc/repofind.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+## Synopsis
+
+repofind [-isu ARG] [--] ARGS...
+
+## Description
+
+## Operations
+
+* __-i__ Find and list all file:// repos in pacman.conf
+* __-s__ Interactively select a repo to use
```
Comment (AladW) [inline doc/repofind.md]: repofind -s gives you a selection dialog, and prints the result as "name
location" on stdout. So, "to use" doesn't make sense in this context (though it does when using repofind -s in other scripts, such as aursync).

File: doc/repoman.md [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+repoman - Run pacman, only considering the official repositories
+
+## Synopsis
+
+repoman ARG
+
+## Description
+
+Repoman is a wrapper for pacman which removes all unofficial repositories from the configuration. All arguments given are directly passed on to pacman
```
Comment (AladW) [inline doc/repoman.md]: Well, this is true in practice. But the inverse, i.e. creating a configuration file
with only the official repositories, reflects the repoman logic more
accurately.

**Comment by AladW** [comment]
Nice! I've added a few comments.

File: doc/aurbuild.md (hunk 1 above)
Comment (runical) [inline doc/aurbuild.md]: Ah, that wasn't clear from my experimenting. Specifying the PKGBUILD also works, hence my confusion. It does explain why I couldn't specify multiple PKGBUILDs though.

File: doc/aurchain.md (hunk 3 above)
Comment (runical) [inline doc/aurchain.md]: That is what I meant, but indeed, this is not a clear description.

File: doc/aurchain.md (hunk 4 above)
Comment (runical) [inline doc/aurchain.md]: Then it is probably best to remove it from the documentation as well (the switch is still in the code though).

File: doc/aursync.md (hunk 8 above)
Comment (runical) [inline doc/aursync.md]: Then I'll remove it.

**Comment by runical** [comment]
I'll get to work on them soon. Any comments additions for aurqueue?

File: doc/aurbuild.md (hunk 1 above)
Comment (AladW) [inline doc/aurbuild.md]: Well, that's odd. Could be that it reads the directory name from the first line (pkgname).

File: doc/aurchain.md (hunk 4 above)
Comment (AladW) [inline doc/aurchain.md]: Removed `-v` with 77e0c3987bde8406a37b37fb6ac57edb84cc9866

**Comment by AladW** [comment]
Well, aurqueue has no options, so you could remove the Operations section. See also could contain datamash(1) and tsort(1) (maybe awk(1)). For the description, it basically does this:
+ Get make/depends from .SRCINFO (paths are retrieved with `find` from CLI arguments)
+ Make "pkgname depends" pairs, and validate them with `datamash`
+ If correct, feed the result to tsort
+ Replace pkgname with pkgbase (using `aursplit`)
+ Match the output for the original arguments

File: doc/aurbuild.md (hunk 1 above)
Comment (runical) [inline doc/aurbuild.md]: I tested it again, but it was because I was already in the directory. My bad!

**Comment by runical** [comment]
I have incorporated your suggestions in the new version. I find especially aurqueue a hard one to aptly describe, but I think this is a decent description.

File: doc/repofind.md [hunk 11]
```diff
@@ -12,10 +12,14 @@ repofind [-isu ARG] [--] ARGS...
 
 ## Operations
 
-* __-i__ Find and list all file:// repos in pacman.conf
-* __-s__ Interactively select a repo to use
+* __-i__ Find and list all file:// (local) repositoroes in pacman.conf
```
Comment (runical) [inline doc/repofind.md]: Crap, spelling error...

File: doc/aurbuild.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+aurbuild - Build packages to a local repository
+
+## Synopsis
+
+aurbuild [+-a ARG] [+-cd ARG] [+-p ARG] [+-r ARG} [--] ARGS...
+
+## Description
+
+Build a package from a queue file and add them to a local repository. A local repository is a repository on the local file system. It is configured using the file:// prefix for Server in pacman.conf.
```
Comment (AladW) [inline doc/aurbuild.md]: I may mention that the queue file includes directories which themselves contain a PKGBUILD.

File: doc/aurbuild.md [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+
+## Operations
+
+* __-a__ A text file with directories containing PKGBUILDs (
```
Comment (AladW) [inline doc/aurbuild.md]: A directory would only include a single PKGBUILD (as aurbuild doesn't use `-p <buildscript>`). Also extraneous "(" at the end.

File: doc/aurbuild.md [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+* __-c__ Build in a chroot
+* __-d__ The name of the database
+* __-p__ Select the pool for the built packages. The pool is the location where aurbuild will put the packages after building
```
Comment (AladW) [inline doc/aurbuild.md]: "Select" is not needed here, as the choice is not interactive.

File: doc/aurbuild.md [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+* __-r__ Select the root for the repository where the database files will live
+
+All arguments after -- get passed to makepkg
```
Comment (AladW) [inline doc/aurbuild.md]: Nitpick: s/get/are. Also, 3e3ba2410389581398bff5fb0d7c60f64c5a9c3c allowed to pass arguments to makechrootpkg, when `-c` is provided.

File: doc/aurchain.md [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+
+aurchain - Solve and order dependencies using the AUR RPC
+
+## Synopsis
+
+aurchain package [package, ...]
+
+## Description
+
+aurchain takes package names and then solves and orders the dependencies for those packages using the AUR RCP interface.
```
Comment (AladW) [inline doc/aurchain.md]: Typo: s/RCP/RPC

File: doc/aurqueue.md [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+
+aurqueue - Order dependencies using .SRCINFO files
+
+## Synopsis
+
+aurqueue pkgbase [pkgbase, ...]
+
+## Description
+
+aurqueue takes the names of packages and validates the dependencies using datamash. It then orders the dependency chains for the given packages.
```
Comment (AladW) [inline doc/aurqueue.md]: I'd focus on tsort, as datamash is an implementation detail (checking if a tsort argument is valid).

File: doc/aursearch.md [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+
+aursearch - Search AUR package names by PCRE pattern
+
+## Synopsis
+
+aursearch [+-Fbqrv} [--] ARGS...
+
+## Description
+
+Search the AUR for packages. The argument is a PCRE pattern to search the AUR with.
```
Comment (AladW) [inline doc/aursearch.md]: Hmm... "search the AUR _with_" sounds a bit odd. Perhaps "to match against the AUR" or so.

File: doc/aursearch.md [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+
+## Operations
+
+* __-F__ Search for a fixed string instead of a PCRE pattern
+* __-b__ Match pkgbase instead of pkgname
+* __-q__ Output less
+* __-r__ Output raw JSON from AUR RPC
```
Comment (AladW) [inline doc/aursearch.md]: Nitpick: "raw JSON" is a tautology, I'd just say "Output JSON from [the] AUR RPC"

File: doc/repoman.md [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+
+repoman - Run pacman, only considering the official repositories
+
+## Synopsis
+
+repoman ARG
+
+## Description
+
+Repoman is a wrapper for pacman which generates a temporary configuration file with only the official repositories. This configuration is then used to run pacman with the arguments passed to repoman.
```
Comment (AladW) [inline doc/repoman.md]: Yup, that puts it very well indeed.

File: doc/aursync.md [hunk 21]
```diff
@@ -0,0 +1,33 @@
...
+
+aursync - download and build AUR packages automatically
+
+## Synopsis
+
+aursync [+-cdi ARG] [+-knut] [--] ARGS...
+
+## Description
+
+Downloads, builds and adds AUR packages automatically to a local repository (local repositories are configured with the file:// prefix). In case there are multiple local repositories configured in pacman.conf, aursync asks which repository to add a package to.
```
Comment (AladW) [inline doc/aursync.md]: I'd consider mentioning aurbuild(1), rather than repeating the details here.

File: doc/aursync.md [hunk 22]
```diff
@@ -0,0 +1,33 @@
...
+
+* __-c__ Build packages in a chroot
+* __-d__ Only download source files, don't build anything
+* __-i__ Ignore package. Currently unused
+* __-k__ Don't view downloaded files before building
+* __-n__ Disable version checking for packages in the queue
+* __-u__ Update packages in the custom repository
+* __-t__ Download the PKGBUILDs using aria instead of git
+
+## See also
```
Comment (AladW) [inline doc/aursync.md]: As aursync wraps other aurutils scripts, I'd mention them here (aurchain, aursearch, aurbuild, repofind, aursift)

**Comment by AladW** [comment]
Great, thanks. Added a few more comments. :+1:

File: doc/aursync.md (hunk 22 above)
Comment (runical) [inline doc/aursync.md]: That makes sense. Done.

**Comment by runical** [comment]
And some more improvements. The main changes are in aursync and aurqueue. Aursync is more concise now and references aurbuild for more information.

**Comment by AladW** [comment]
One last nitpick would be to change "A text file with directories containing _a single PKGBUILD to build_" to "A text file with directories containing _a PKGBUILD_".

Otherwise this should be ready to be merged. Making it groff would be done by hand, as "converters" like pod2man and pandoc make a mess (see https://ptpb.pw/c12l for pacsift(1), which uses pod2man). I wouldn't mind doing this, but if you want to give it a shot, this might be better done in a separate pull request.

**Comment by runical** [comment]
All right, your final nitpick has been addressed. If it isn't too much trouble, I'll leave the conversion to you. Happy to help with the text though!

**Comment by AladW** [comment]
Merged, thanks for the help!

**Pull request merged**